packages/nvidia_nat_core/src/nat/middleware/defense/defense_middleware_pre_tool_verifier.py

```
@@ -0,0 +1,406 @@
+        """
+        logger.warning("Pre-Tool Verifier detected violation in input to %s: %s (confidence=%s, types=%s)",
+                       context.name,
+                       analysis_result.reason,
+                       analysis_result.confidence,
+                       analysis_result.violation_types)
+
+        action = self.config.action
+
+        if action == "refusal":
+            logger.error("Pre-Tool Verifier refusing input to %s: %s", context.name, analysis_result.reason)
+            raise ValueError("Input blocked by security policy")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)